src/scripts/main.js

```diff
@@ -1 +1,67 @@
 'use strict';
+
+new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    // eslint-disable-next-line prefer-promise-reject-errors
+    reject('First promise was rejected');
+  }, 3000);
+
+  document.body.addEventListener(
+    'click',
+    (e) => {
+      resolve('First promise was resolved');
+      clearTimeout(timeoutId);
+    },
+    { once: true },
+  );
+})
+  .then((successResults) => {
+    createNotification(successResults, true);
+  })
+  .catch((errorResult) => {
+    createNotification(errorResult, false);
+  });
+
+new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
+  });
+}).then((successResults) => {
+  createNotification(successResults, true);
+});
+
+new Promise((resolve) => {
+  let left = false;
+  let right = false;
+
+  document.addEventListener('click', () => {
+    left = true;
+    checkCondition();
+  });
+
+  document.addEventListener('contextmenu', () => {
+    right = true;
+    checkCondition();
+  });
+
+  function checkCondition() {
+    if (left && right) {
+      resolve('Third promise was resolved');
+    }
+  }
+}).then((successResults) => {
+  createNotification(successResults, true);
+});
```

Comment:
For the third Promise, you are using two flags to track 'click' and 'contextmenu' events. However, the same issue as the second Promise persists here, the event listeners are not removed after the Promise is resolved. It's better to remove event listeners once the Promise has been resolved.